myproject/supermarket_manager/views.py

```diff
@@ -384,3 +388,39 @@ def get_account_detail(request, account_id):
         return JsonResponse(account_data, status=200)
     except Account.DoesNotExist:
         return JsonResponse({"message": NOT_FOUND}, status=404)
+
+
+@csrf_protect
+def login_account(request):
+    """
+    API endpoint for user login.
+
+    This function handles POST requests for user login. Users can log in using their email
+    and password. If either the email or password is missing, it returns a 400 Bad Request
+    response. If the login credentials are incorrect, it returns a 401 Unauthorized response.
+
+    Parameters:
+        request (HttpRequest): The HTTP request object containing user login credentials.
+
+    Returns:
+        JsonResponse: A JSON response indicating the result of the login attempt.
+    """
+
+    if request.method == "POST":
```

Comment:
Limit nesting of conditional statements.

Reply:
Done